ruby/gilded_rose_tests.rb

```diff
@@ -2,11 +2,124 @@
 require 'test/unit'
 
 class TestUntitled < Test::Unit::TestCase
+  MAX_QUALITY = 50
 
-  def test_foo
-    items = [Item.new("foo", 0, 0)]
-    GildedRose.new(items).update_quality()
-    assert_equal items[0].name, "fixme"
+  test "Once the sell by date has passed, quality degrades twice as fast" do
+    initial_expired_item_quality = 2
+    initial_unexpired_item_quality = 2
+
+    expired_item = Item.new(name = "TestItem", sell_in = 0, quality = initial_expired_item_quality)
+    unexpired_item = Item.new(name = "TestItem", sell_in = 2, quality = initial_unexpired_item_quality)
+
+    items = [expired_item, unexpired_item]
+    GildedRose.new(items).update_quality
+
+    assert_equal initial_expired_item_quality - 2, expired_item.quality
+    assert_equal initial_unexpired_item_quality - 1, unexpired_item.quality
+  end
+
+  test "Once the sell by date has passed, quality degrades but it does not drop below zero" do
+    expired_item = Item.new(name = "TestItem", sell_in = 0, quality = 0)
+
+    gilded_rose = GildedRose.new([expired_item])
+
+    gilded_rose.update_quality
+    assert_equal 0, expired_item.quality
+  end
+
+  test "After every update_quality call, sell_in is reduced by 1" do
+    number_of_updates = 10
+    initial_item_sell_in = 4
+
+    item = Item.new(name = "TestItem", sell_in = initial_item_sell_in, quality = 0)
+    gilded_rose = GildedRose.new([item])
+    number_of_updates.times { gilded_rose.update_quality }
+
+    expected_item_sell_in = initial_item_sell_in - number_of_updates
+    assert_equal expected_item_sell_in, item.sell_in
```

Comment:
`assert_change` could also be used. We should check to see if `assert_change` supports `by: `.

```ruby
assert_change(-> { expired_item.quality }, by: number_of_updates) do
  gilded_rose.update_quality
end
```